new-lamassu-admin/src/pages/Analytics/graphs/TopMachinesBarGraph.js

```diff
@@ -0,0 +1,321 @@
+
+  // This ensures that the graph renders a minimum amount of machines
+  // and avoids having a single bar for cases with one machine
+  const filledMachines =
+    R.length(machines) >= AMOUNT_OF_MACHINES
+      ? machinesClone
+      : R.map(it => {
+          if (!R.isNil(machinesClone[it])) return machinesClone[it]
+          return { code: `ghostMachine${it}`, display: `` }
+        }, R.times(R.identity, AMOUNT_OF_MACHINES))
+
+  const txByDevice = R.reduce(
+    (acc, value) => {
+      acc[value.code] = R.filter(it => it.deviceId === value.code, data)
+      return acc
+    },
+    {},
+    filledMachines
+  )
+
+  const getDeviceVolume = deviceId =>
+    R.reduce(
+      (acc, value) => acc + BigNumber(value.fiat).toNumber(),
+      0,
+      txByDevice[deviceId]
+    )
+
+  const getDeviceVolumeByTxClass = deviceId =>
+    R.reduce(
+      (acc, value) => {
+        if (value.txClass === 'cashIn')
+          acc.cashIn += BigNumber(value.fiat).toNumber()
+        if (value.txClass === 'cashOut')
+          acc.cashOut += BigNumber(value.fiat).toNumber()
+        return acc
+      },
+      { cashIn: 0, cashOut: 0 },
+      txByDevice[deviceId]
+    )
+
+  const devicesByVolume = R.sort(
+    (a, b) => b[1] - a[1],
+    R.toPairs(
+      R.reduce(
+        (acc, value) => {
+          acc[value.code] = getDeviceVolume(value.code)
+          return acc
+        },
+        {},
+        filledMachines
+      )
+    )
+  )
+
+  const topMachines = R.take(AMOUNT_OF_MACHINES, devicesByVolume)
+
+  const txClassVolumeByDevice = R.reduce(
+    (acc, value) => {
+      acc[value[0]] = getDeviceVolumeByTxClass(value[0])
+      return acc
+    },
+    {},
+    topMachines
+  )
+
+  const x = d3
+    .scaleBand()
+    .domain(topMachines)
+    .range([GRAPH_MARGIN.left, GRAPH_WIDTH - GRAPH_MARGIN.right])
+    .paddingInner(BAR_PADDING)
+
+  const y = d3
+    .scaleLinear()
+    .domain([
+      0,
+      d3.max(topMachines, d => d[1]) !== 0 ? d3.max(topMachines, d => d[1]) : 50
+    ])
+    .range([GRAPH_HEIGHT - GRAPH_MARGIN.bottom, GRAPH_MARGIN.top])
+
+  const buildXAxis = useCallback(
+    g =>
+      g
+        .attr('class', 'x-axis-1')
+        .attr(
+          'transform',
+          `translate(0, ${GRAPH_HEIGHT - GRAPH_MARGIN.bottom})`
+        )
+        .call(
+          d3
+            .axisBottom(x)
+            .tickFormat(
+              d =>
+                `${R.find(it => it.code === d[0], filledMachines).display ??
+                  ''}`
+            )
+            .tickSize(0)
+            .tickPadding(10)
+        ),
+    [GRAPH_MARGIN, x, filledMachines]
+  )
+
+  const buildXAxis2 = useCallback(
+    g => {
+      g.attr('class', 'x-axis-2')
+        .attr(
+          'transform',
+          `translate(0, ${GRAPH_HEIGHT - GRAPH_MARGIN.bottom})`
+        )
+        .call(
+          d3
+            .axisBottom(x)
+            .tickFormat(d =>
+              R.includes(`ghostMachine`, d[0]) ? `` : `${d[1]} ${currency}`
```

Comment:
I think that we should use only first two decimal places. Or maybe no decimal places at all?

```suggestion
              R.includes(`ghostMachine`, d[0])
                ? ``
                : `${d[1].toFixed(2)} ${currency}`
```